### PR TITLE
Add preliminary support for other OS Support

### PR DESCRIPTION
**Pull request by rspavel**
Refactored existing files to ubuntu specific notation and added example of docker container for fedora as precursor to additional OS support and as example of needs for users building proxy catalog on a fresh fedora install.

Currently the CI tests are not enabled due to concerns over job count and issues with spack's usage of the patch utility in a non-interactive Fedora container.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -70,7 +70,7 @@ before_install:
 script:
 - docker build 
     --build-arg APP="${APP}" --build-arg DEPS="${DEPS}" --build-arg INSECURE="${INSECURE}"
-    -t ${TRAVIS_REPO_SLUG}:${APP} ${HOME}/docker/ &&
+    -t ${TRAVIS_REPO_SLUG}:${APP} ${HOME}/docker/ --file="${HOME}/docker/Dockerfile-${DOCKEROS}" &&
```
Comment (junghans) [inline .travis.yml]: Add ${DOCKEROS} is the Tag here, too!

File: docker/Dockerfile-fedora [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+USER root
+RUN chown -R ci:ci /home/ci/proxy-ci /home/ci/.ccache /home/ci/spack
+USER ci
+
+WORKDIR /home/ci
+RUN cp proxy-ci/fedora-packages.yaml spack/etc/spack/packages.yaml
+ENV SPACK=/home/ci/spack/bin/spack
+ENV SPACK_ROOT=/home/ci/spack
+
+RUN source ${SPACK_ROOT}/share/spack/setup-env.sh
```
Comment (junghans) [inline docker/Dockerfile-fedora]: source doesn’t work in docker!

File: docker/Dockerfile-fedora [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+RUN ${SPACK} install -j4 patch
```
Comment (junghans) [inline docker/Dockerfile-fedora]: Huh? Why?

Comment (rspavel) [reply docker/Dockerfile-fedora]: That is related to ongoing issues with debugging what is different between the docker and interactive environments for the Fedora container with a subset of the proxies that is still being debugged on a different branch prior to even considering enabling those tests

**Comment by rspavel** [comment]
Fixed issues with fedora (apparently bzip2 and possibly patch aren't a part of the fedora image) and doubled the number of tests. All of which pass with some manual restarting when travis presumably hammers the package mirrors too hard.

**Review by junghans — APPROVED** [review]

**Pull request merged**